pkg/medusa/reconcile.go

```diff
@@ -187,6 +237,116 @@ func UpdateMedusaMainContainer(dcConfig *cassandra.DatacenterConfig, medusaSpec
 	}
 }
 
+func generateMedusaProbe(configuredProbeSettings *api.ProbeSettings) *corev1.Probe {
```

Comment:
Could/should this be done via Goalesce? Note that I can't see `TerminationGracePeriodSeconds` in here, which may be an important configurable in some cases.